src/components/Axis/HorizontalAxis.tsx

```diff
@@ -28,38 +35,55 @@ export default function HorizontalAxis(props: AxisRendererProps) {
   const transform = isBottom ? `translate(0, ${plotHeight})` : undefined;
 
   const ticks = useBBoxObserver();
-
-  function getTickPosition(value: number) {
-    return {
-      line: {
-        x1: value,
-        x2: value,
-        y2: !isBottom ? -5 : 5,
-      },
-      text: { x1: value, y1: isBottom ? 16 : -12 },
+  function getTickPosition(
+    position: 'inner' | 'outer' | 'center',
+    primaryTickLength: number,
+  ) {
+    const y1 = position === 'center' ? (isBottom ? -5 : primaryTickLength) : 0;
```

Comment:
There's some confusion in this function:
- when position is `inner`, the tick should be drawn inside the graph (below the top axis or above the bottom axis)
- when position is `outer`, the tick should be drawn outside the graph (above the top axis or below the bottom axis)
- when position is `center`, the tick should be in between (shifted by `primaryTickLength / 2`)

The text coordinates should also depend on the tick length and position of the ticks, instead of being hardcoded to 16/-12

Reply:
fixed